Podcast/ExternalProfileViewController.swift

```diff
@@ -309,7 +309,9 @@ class ExternalProfileViewController: ViewController, UITableViewDataSource, UITa
     }
     
     func recommendedSeriesTableViewCell(cell: RecommendedSeriesTableViewCell, didSelectItemAt indexPath: IndexPath) {
-        let seriesDetailViewController = SeriesDetailViewController(series: subscriptions[indexPath.row])
+        let seriesDetailViewController = SeriesDetailViewController()
+        // fetch new series because isSubscribed is different for current user
+        seriesDetailViewController.fetchSeries(seriesID: subscriptions[indexPath.row].seriesId)
```

Comment:
Hmm we shouldn't need to fetch it all over again. The original fetch should have this data already.

Reply:
And if not it needs to be added, we shouldn't be setting this locally on fetch.

Reply:
We actually need this call and the result of not calling it was the reason for the bug. Since this is the external profile that means the series isSubscribed is considered the isSubscribed of the external profile user and not the currentUser. Thus, we need to make this call to fetch the series again or else we won't have that isSubscribed currently reflecting the currentUser. One thing we could do is get it from the cache right. Well in the cache its stored with the isSubscribed state of the external profile user and not the current user. So when we request the series again it will update the cache for this series. We could also call a isUserSubscribed endpoint but unfortanetly no such endpoint exists. Even if one did the only thing we would probably be saving here is a couple bits over the network.